avocado/utils/lv_utils.py

```diff
@@ -356,8 +366,50 @@ def lv_create(vg_name, lv_name, lv_size, force_flag=True):
     elif lv_check(vg_name, lv_name) and force_flag:
         lv_remove(vg_name, lv_name)
 
-    cmd = "lvcreate --size %s --name %s %s -y" % (lv_size, lv_name, vg_name)
-    process.run(cmd, sudo=True)
+    lv_cmd = "lvcreate --name %s" % lv_name
+    if pool_name is not None:
+        if not lv_check(vg_name, pool_name):
+            tp_cmd = "lvcreate --thinpool %s --size %s %s -y" % (pool_name,
+                                                                 pool_size,
+                                                                 vg_name)
+            try:
+                process.run(tp_cmd, sudo=True)
+            except process.CmdError as detail:
+                LOGGER.debug(detail)
+                raise LVException("Create thin volume pool failed.")
+            LOGGER.debug("Created thin volume pool: %s", pool_name)
+        lv_cmd += " --virtualsize %s" % lv_size
+        lv_cmd += " --thin %s/%s -y" % (vg_name, pool_name)
+    else:
+        lv_cmd += " --size %s" % lv_size
+        lv_cmd += " %s -y" % vg_name
+    try:
+        process.run(lv_cmd, sudo=True)
+    except process.CmdError as detail:
+        LOGGER.error(detail)
+        raise LVException("Create thin volume failed.")
+    LOGGER.debug("Created thin volume:%s", lv_name)
+
+
+def thin_lv_create(vg_name, thinpool_name="lvthinpool", thinpool_size="1.5G",
```

Comment:
The `thinpool_size` default here is 1.5G but it will become 1G when we drop this function (as it's marked as deprecated). So, should the default `pool_size` on `lv_create` be 1.5G too?

Reply:
Right. I'm going to proceed with the merge.